### PR TITLE
feat(PPDSC2091): caption refactoring

### PR DESCRIPTION
**Pull request by mutebg**
PPDSC-2091

**What**

What did you do:
- removed CaptionInset
- refactored Caption to use GridLayout
- Added codemod which replaces CaptionInset with Caption with padding.

**I have done:**
- [ ] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [x] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by pp-serviceaccount** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2091-caption-v2
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2091-caption-v2/storybook

File: lib/codemod/src/utils/replace-space-inset.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
+const {isObject} = require('lodash');
```
Comment (Xin00163) [inline lib/codemod/src/utils/replace-space-inset.js:1]: can we use the `isObject` fun from utils/helper?
Save us from install lodash

**Review by Xin00163 — COMMENTED** [review]
"The caption can be inset with increased spacing. There are two sizes of inset: XS and MD." This probably needs an update.

Is there any updates on types.ts?

File: lib/codemod/package.json [hunk 2]
```diff
@@ -22,6 +22,7 @@
     "glob": "^8.0.3",
     "jscodeshift": "^0.13.1",
     "jscodeshift-add-imports": "^1.0.10",
+    "lodash": "^4.17.21",
```
Comment (Vanals) [inline lib/codemod/package.json:25]: Should this be in devDependencies?

File: lib/codemod/src/utils/replace-space-inset.js (hunk 1 above)
Comment (mutebg) [reply lib/codemod/src/utils/replace-space-inset.js:1]: I forgot to update the types.
Also removed the lodash for now.

Thanks

File: lib/codemod/package.json (hunk 2 above)
Comment (mutebg) [reply lib/codemod/package.json:25]: I removed it. its not needed for now

**Review by JohnTParsons — APPROVED** [review]

**Review by Xin00163 — APPROVED** [review]

**Pull request merged**